docs/training-debug/index.txt

```diff
@@ -1,343 +1,397 @@
-to ``det experiment create`` would.
+   .. code:: bash
 
-**How to diagnose failures:** Failure diagnosis is a combination of the failure diagnosis for Step 3
-and Step 4.
+      det shell start --context my_model_dir
+      # Your terminal should automatically connect to the shell.
 
-*********************
- Higher-level Issues
-*********************
+   When interacting with the shell or notebook, testing is the same as :ref:`Step 2 <step2>`.
 
-7. Does cluster test mode work with ``slots_per_trial`` set to ``1``?
-=====================================================================
+#. Diagnose failures:
 
-This step is conceptually similar to Step 6, except instead of launching the command from an
-interactive environment, we will submit it to the cluster and let Determined manage everything.
+   Combine the failure diagnosis steps used in :ref:`Step 2 <step2>` and :ref:`Step 4 <step4>`.
 
-**How to test:** If you had to make any customizations to your command environment while testing
-Steps 3, 4, or 5, make sure that you have made the same customizations in your experiment config.
-Then also confirm that your experiment config either does not specify ``resources.slots_per_trial``
-at all, or that it is set to 1, like:
+.. _step6:
 
-.. code:: yaml
+*******************************************************************
+ Step 6 - Verify that local test mode works in a notebook or shell
+*******************************************************************
 
-   resources:
-     slots_per_trial: 1
+This step is the same as :ref:`Step 3 <step3>`, except the original code runs on the Determined
+cluster instead of locally.
 
-Then create an experiment with the ``--test`` flag (but not the ``--local`` flag):
+#. Launch a notebook or shell as described in :ref:`Step 4 <step4>`.
 
-.. code:: bash
+   On the cluster, testing is the same as :ref:`Step 3 <step3>`, except that the second model
+   definition argument of the ``det experiment create`` command should be
+   ``/run/determined/workdir`` or ``.`` if you have not changed the working directory after
+   connecting to the cluster. This is because the ``--context`` specified when creating the shell or
+   notebook is copied to the ``/run/determined/workdir`` directory inside the container, the same as
+   the model definition argument is copied to ``det experiment create``.
 
-   det experiment create myconfig.yaml my_model_dir --test
+#. Perform failure diagnosis:
 
-**How to diagnose failures:** If you were able to run local test mode inside a notebook or shell,
-but you are unable to successfully submit an experiment, you should focus on making sure that any
-customizations you made to get it to work in the notebook or shell have been properly replicated in
-your :ref:`experiment config <experiment-configuration>`:
+   Follow the combined failure diagnostic steps used for :ref:`Step 3 <step3>` and :ref:`Step 4
```

Comment:
suggested reword: "Follow the combined failure diagnoses from..."

Similar motivation as "Diagnose failures" vs "Perform failure diagnosis".  I don't feel strongly about this reword at all.

Reply:
Fixed.